tests/suites/test_suite_pk.function

```diff
@@ -1835,18 +1837,18 @@ void pk_psa_sign(int psa_type, int bits, int rsa_padding)
     USE_PSA_INIT();
 
     /* Create the legacy EC/RSA PK context. */
-#if defined(MBEDTLS_RSA_C) && defined(MBEDTLS_GENPRIME)
+#if defined(MBEDTLS_RSA_C)
     if (PSA_KEY_TYPE_IS_RSA(psa_type)) {
-        TEST_EQUAL(pk_genkey(&pk, mbedtls_pk_info_from_type(MBEDTLS_PK_RSA), bits), 0);
+        TEST_EQUAL(pk_genkey(&pk, MBEDTLS_PK_RSA, bits), 0);
         TEST_EQUAL(mbedtls_rsa_set_padding(mbedtls_pk_rsa(pk), rsa_padding, MBEDTLS_MD_NONE), 0);
     }
-#else /* MBEDTLS_RSA_C && MBEDTLS_GENPRIME */
+#else /* MBEDTLS_RSA_C && MBEDTLS_PK_PARSE_C */
```

Comment:
Doesn't match the opening guard, should be just `RSA_C`. Same for `#endif` two lines below.